File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/entity/impl/BitmapEntityManager.java
package com.dkellycollins.triggerfinger.managers.entity.impl;

import android.graphics.Bitmap;

import com.dkellycollins.triggerfinger.data.daos.IBitmapDao;
import com.dkellycollins.triggerfinger.managers.entity.IBitmapEntityManager;
import com.dkellycollins.triggerfinger.util.qa.Assert;

public class BitmapEntityManager implements IBitmapEntityManager {

    private final IBitmapDao _dao;

    public BitmapEntityManager(IBitmapDao dao) {
        Assert.isNotNull(dao, "dao");

        _dao = dao;
    }

    @Override
    public Bitmap retrieve(int id) {
        return _dao.retrieve(id);
    }

    @Override
    public void load(int id) {
        _dao.loadBitmap(id);
    }

    @Override
    public void delete(int id) {
        _dao.delete(id);
    }
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/entity/impl/BitmapEntityManager.kt
package com.dkellycollins.triggerfinger.managers.entity.impl

import android.graphics.Bitmap

import com.dkellycollins.triggerfinger.data.daos.IBitmapDao
import com.dkellycollins.triggerfinger.managers.entity.IBitmapEntityManager

class BitmapEntityManager(private val _dao: IBitmapDao) : IBitmapEntityManager {

    override fun retrieve(id: Int): Bitmap {
        return _dao.retrieve(id)
    }

    override fun load(id: Int) {
        _dao.loadBitmap(id)
    }

    override fun delete(id: Int) {
        _dao.delete(id)
    }
}
